src/Chainweb/Pact/TransactionExec.hs

```diff
@@ -586,22 +581,56 @@ initStateInterpreter :: EvalState -> Interpreter p
 initStateInterpreter s = Interpreter $ \start end withRollback runInput ->
     withRollback (put s >> start runInput >>= end)
 
+checkTooBigTx
+    :: Gas
+    -> GasLimit
+    -> TransactionM p (CommandResult [TxLog Value])
+    -> TransactionM p (CommandResult [TxLog Value])
+checkTooBigTx initialGas gasLimit next
+  | initialGas >= (fromIntegral gasLimit) = do
+      txGasUsed .= (fromIntegral gasLimit) -- all gas is consumed
+
+      let !pe = PactError GasError def []
+            $ "Tx too big (" <> pretty initialGas <> "), limit "
+            <> pretty gasLimit
+
+      jsonErrorResult pe "Tx too big"
+  | otherwise = next
+
+gasInterpreter :: Gas -> TransactionM db (Interpreter p)
+gasInterpreter g = do
+    mc <- use txCache
+    return $ initStateInterpreter
+        $ set evalGas g
+        $ setModuleCache mc def
+
 -- | Initial gas charged for transaction size
 --   ignoring the size of a continuation proof, if present
 --
 initialGasOf :: PayloadWithText -> Gas
 initialGasOf cmd = gasFee
   where
     feePerByte :: Decimal = 0.01
-
+    
     contProofSize =
       case _pPayload (payloadObj cmd) of
         Continuation (ContMsg _ _ _ _ (Just (ContProof p))) -> B.length p
         _ -> 0
     txSize = SB.length (payloadBytes cmd) - contProofSize
-    gasFee = round $ fromIntegral txSize * feePerByte
+
+    costPerByte = fromIntegral txSize * feePerByte
+    sizePenalty = txSizeAccelerationFee costPerByte
+    gasFee = ceiling (costPerByte + sizePenalty)
 {-# INLINE initialGasOf #-}
 
+txSizeAccelerationFee :: Decimal -> Decimal
+txSizeAccelerationFee costPerByte = total
+  where
+    total = (costPerByte / bytePenalty) ^ power
```

Comment:
Where did the factor go? I thought it was (s * F) + ((s / P) ^ 7)

Reply:
I thought the formula was 
```
(s * F) + (((s * F) / 512) ^ 7)
```
and 
```
costPerByte = s * F
```

Reply:
ok got it